### ISSUE TITLE
my raw_data are in the form of .obj files，how can I use your work but trained with my own data? 

### ISSUE THREAD

**Issue by ZhangRuizhe2002** [Tai-Hsien/MeshSegNet#40]

**Comment by ZhangRuizhe2002** [comment]
Another question, my raw_data are in huge amount of points.How can I downsample them before I train

**Comment by ZhangRuizhe2002** [comment]
These are some of my samples ,I will be very grateful if you can help!![https://cloud.tsinghua.edu.cn/d/c7901b29d37b437d8b22/](url)

**Comment by ZhangRuizhe2002** [comment]
Oh,sorry to bother again.I have no idea about merging my lable and data into one vtp files,like you did in the project.I will be very grateful if you can help!
Best regards

**Comment by Tai-Hsien** [comment]
> These are some of my samples ,I will be very grateful if you can help!![https://cloud.tsinghua.edu.cn/d/c7901b29d37b437d8b22/](url)

I checked your sample and I believe you use the same way to save the label information as MICCAI2022 3D TeethSeg challenge. 
Here are my suggestion.
(1) Convert the "label list" in the json file into a vedo point array. For more information about vedo, please check https://vedo.embl.es/.
(2) We use cell/triangle as a basic unit in our previous work; however, the data in 3DTeethSeg challenge use vertex/point as a unit. The simplest way to convert the label in vertex basis into cell basis is to use KNN, where you use all vertex location and vertex label as training samples, train a KNN model, and predict the cell label when feeding cell location.

**Comment by Tai-Hsien** [comment]
> > These are some of my samples ,I will be very grateful if you can help!![https://cloud.tsinghua.edu.cn/d/c7901b29d37b437d8b22/](url)
> 
> I checked your sample and I believe you use the same way to save the label information as MICCAI2022 3D TeethSeg challenge. Here are my suggestion. (1) Convert the "label list" in the json file into a vedo point array. For more information about vedo and learn how to use vedo, please check https://vedo.embl.es/. (2) We use cell/triangle as a basic unit in our previous work; however, the data in 3DTeethSeg challenge use vertex/point as a unit. The simplest way to convert the label in vertex basis into cell basis is to use KNN, where you use all vertex location and vertex label as training samples, train a KNN model, and predict the cell label when feeding cell location.

**Comment by ZhangRuizhe2002** [comment]
Thanks for your early reply! Actually,I am taking this 3DTeethSeg Challenge.But it is hard for me without baseline.Could you please help me with a baseline which I can run myself.
Best regards!

**Comment by Tai-Hsien** [comment]
Once you implement the two things that I suggest, you can convert the data to VTP file with cell label and implement the code we provide to train it. If you don't know how to convert VTP, vedo website will be a good start.

**Comment by ZhangRuizhe2002** [comment]
OK. I will try.Thanks a lot!

**Comment by ZhangRuizhe2002** [comment]
Besides,I wonder how to transform points to cells.I've found no function in the vedo website

**Comment by Tai-Hsien** [comment]
> The simplest way to convert the label in vertex basis into cell basis is to use KNN, where you use all vertex location and vertex label as training samples, train a KNN model, and predict the cell label when feeding cell location.

See the second point in my previous comment.

The simplest way to convert the label in vertex basis into cell basis is to use KNN, where you use all vertex location and vertex label as training samples, train a KNN model, and predict the cell label when feeding cell location.

To implement KNN, you can use sklearn.

**Comment by ZhangRuizhe2002** [comment]
Sorry.I didn't express my question clearly.I don't know the differences between cells and points.My obj files have many points and vertices,but I don't know how to change it into cells. Are cells just points with points?

**Comment by Tai-Hsien** [comment]
In mesh data, a 3D object consists of many points and triangles.
Point = vertex
Cell = triangle, so each cell has three points.
In their json file, they provide a label for each point, so you need to convert the labels of points to labels of cells.

**Comment by ZhangRuizhe2002** [comment]
Thanks!!!
